src/editor.js

```diff
@@ -111,11 +174,68 @@ CKEDITOR.define( [
 		 * element will be recovered.
 		 *
 		 * @fires destroy
+		 * @returns {Promise} A promise that resolves once the editor instance is fully destroyed.
 		 */
 		destroy: function() {
+			var that = this;
+
 			this.fire( 'destroy' );
 
-			delete this.element;
+			this._destroyPromise = this._destroyPromise || Promise.resolve()
```

Comment:
There is missing test for it. I can remove `this._destroyPromise ||` and tests are still green. This is easy feature which can be easily tested and event easier skipped during future refactoring so it needs tests.